src/app/pages/SearchFunction.ts

```diff
@@ -0,0 +1,199 @@
+import { Result } from '../../__types__';
+
+const MAX_RESULT = 20; // stop searching once we get more than 10 results
+const MAX_TEXT_LENGTH = 256; // get this many text as a preview for
+
+// return
+// * a string at most <MAX_TEXT_LENGTH> chars long, centered around the keyword position
+// * a boolean value, indicating if this is a placeholder text because the keyword
+//   was not found in the raw text due to some indexing issue
+// A placeholder text maybe replaced later by some other non-placeholder text
+function getShortText(keyword: string, url: string, siteMapDatabase: any): [string, boolean] {
+    let fullText: string = siteMapDatabase[url].text;
+    if (!fullText) {
+        return ['', true];
+    }
+    fullText = fullText
+        .replace(/import .*? from .*?;/gim, '')
+        .replace(/\r\n/g, '\n')
+        .replace(/\n/gim, ' ')
+        .replace(/<!--.*?-->/g, ' ')
+        .replace(/\[(.*?)\]\(.*?\)/g, '$1') // replace all the markdown links [text](url) into text
+        .replace(/<[a-z].*?>/gim, ' ')
+        .replace(/[#`]/g, '')
+        .replace(/\s\s+/g, ' '); // replace multiple spaces into only one space
```

Comment:
didn't we already do something like this when we created the DB?

Reply:
when we index them, yes. Not "when we crawl them" yet. I'd think about what to do with it.

Reply:
Nah, I will just leave them as they are. The text "sanitizer" rules are slightly different for the indexing stage, parse query stage, and fetch short text stage. Hard to concat everything into one function.